### ISSUE TITLE
Mismatch between Sides and size of Vertex - fatal error in v9.0

### ISSUE THREAD

**Issue by mjwitte** [NREL/EnergyPlus#6995]
Issue overview
--------------
User file which runs in v8.9, fails in v9.0 with errors like this:
```
   ** Warning ** GetVertices: Distance between two vertices < .01, possibly coincident. for Surface=SURFACE 8536, in Zone=THERMAL ZONE 142
   **   ~~~   ** Vertex [12]=(-1.94,64.72,21.40)
   **   ~~~   ** Vertex [1]=(-1.94,64.72,21.40)
   **   ~~~   ** Dropping Vertex [12].
   ** Severe  ** BuildingSurface:Detailed="SURFACE 8536", After CheckConvexity, mismatch between Sides (11) and size of Vertex (12).
```
Looks like the vertex size is not getting reduced when a coincident vertex is dropped. Likely related to #6853.

### Details
Some additional details for this issue (if relevant):
 - Platform Win64
 - Version of EnergyPlus 9.0.0-2ef880da82
 - Helpdesk ticket number 13067
### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Defect file added to EnergyPlusDevSupport\DefectFiles - *however, the user files are enormous (approx 600 zones), so you can't even see the v8.9 file finish successfully.  But the 9.0 file fails quickly. Working on a smaller file to reproduce the same issue.*
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

**Comment by mjwitte** [comment]
The plot thickens. 
1. Made a smaller version of the original file with just one zone that includes one of the failing surfaces. Ran fine in both v8.9 and v9.0.
2. Noted that the failing surface was originally an interzone surface, so added a second zone and the paired surface - now it runs in v8.9 and fails in v9.0.
Smaller defect file attached here.


[6995-SmallerDefectFile.idf.txt](https://github.com/NREL/EnergyPlus/files/2448410/6995-SmallerDefectFile.idf.txt)
[6995-SmallerDefectFile_V890.idf.txt](https://github.com/NREL/EnergyPlus/files/2448411/6995-SmallerDefectFile_V890.idf.txt)
[6995-SmallerDefectFile-SingleZone.idf.txt](https://github.com/NREL/EnergyPlus/files/2448430/6995-SmallerDefectFile-SingleZone.idf.txt)

**Comment by Myoldmopar** [comment]
Addressed via #6996
